### PR TITLE
Move SwapChain methods from GPUDevice to GPUCanvasContext

### LINKED ISSUE #250: Move createSwapChain from GPUDevice to GPUCanvasContext
Creating a swap chain on the device doesn't make it clear from a web developer point of view that this is going to invalidate the previous one created on a context. I believe moving `createSwapChain` from GPUDevice to GPUCanvasContext would help a lot removing that potential confusion.

```js

const canvas = document.querySelector('canvas');
const context = canvas.getContext('gpupresent');

// Before

const swapChainDescriptor = { context: context, format: 'bgra8unorm' };
const swapChain = device.createSwapChain(swapChainDescriptor);

const format = await device.getSwapChainPreferredFormat(context);

// After

const swapChainDescriptor = { device: device, format: 'bgra8unorm' };
const swapChain = context.createSwapChain(swapChainDescriptor);

const format = await context.getSwapChainPreferredFormat(device);
```

### PR DESCRIPTION
**Pull request by beaufortfrancois**
FIX #250

### CODE REVIEW

File: design/sketch.webidl [hunk 1]
```diff
@@ -687,11 +687,16 @@ interface GPUQueue {
 };
 
 // SwapChain / CanvasContext
-interface GPUCanvasContext {
+partial interface GPUCanvasContext {
+    // Calling createSwapChain a second time invalidates the previous one, an
+    // all of the textures it’s produced.
```
Comment (RafaelCintron) [inline design/sketch.webidl]: Nit: "an all of the textures it's produced" --> "and all of the textures it's produced".

Comment (beaufortfrancois) [reply design/sketch.webidl]: Fixed. Thank you!

File: design/sketch.webidl [hunk 2]
```diff
@@ -687,11 +687,16 @@ interface GPUQueue {
...
+    GPUSwapChain createSwapChain(GPUSwapChainDescriptor descriptor);
```
[the remark this answers is not in the document]
Comment (beaufortfrancois) [reply design/sketch.webidl]: `s/createSwapChain/configureSwapChain/`. Thanks!

**Comment by beaufortfrancois** [comment]
I've updated PR based on your feedback.

**Comment by Kangz** [comment]
Great! Merging this.

**Pull request merged**